Source/SwiftLintBuiltInRules/Rules/Lint/UnneededSynthesizedInitializerRule.swift

```diff
@@ -0,0 +1,252 @@
+// swiftlint:disable file_header
+//
+// Adapted from swift-format's UseSynthesizedInitializer.swift
+//
+// https://github.com/apple/swift-format
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
```

Comment:
My gut feeling says this header provides all the necessary information the licence text requires in case of redistribution.

Let @jpsim confirm or object.

Reply:
Yes, I think this is fine. We've done this before with operator folding which we had taken from swift-syntax at some point, but now just use it directly.